### PR TITLE
Set /metrics/ return type to List[str] and add it to the client

### PR DESCRIPTION
**Pull request by samredai**
### Summary

This PR makes a couple of small changes across the client, server, and docs site.
- Set the `/metrics/` endpoint to return only metric names (`List[str]`)
- Add a `dj.metrics()` method to the client that calls `/metrics/`
- Fill in the "Listing Metrics" docs page
- Adds an intro illustration to the README and docs main page

### Test Plan

`make check` && `make test` as well as install the client and trying it against a local DJ server.

- [x] PR has an associated issue: #296 
- [ ] `make check` passes
- [ ] `make test` shows 100% unit test coverage

### Deployment Plan

N/A

### CODE REVIEW

**Review by shangyian — APPROVED** [review]
Goodbye latency 🤞

File: docs/static/datajunction-illustration.png
Comment (agorajek) [inline docs/static/datajunction-illustration.png:1]: Awesome pic. What do you think about changing the 3 charts on the right hand side to:
- one graph / BI tool
- one Python Jupyter notebook
- one ChatGPT window

To show that there are many ways to consume the DJ metadata?

File: client/python/datajunction/client.py [hunk 1]
```diff
@@ -304,6 +304,12 @@ def engines(self):
         response = self._session.get("/engines/", timeout=self._timeout)
         return response.json()
 
+    def metrics(self):
```
Comment (agorajek) [inline client/python/datajunction/client.py:307]: Nice and scary. Wonder when we hit some perf issues when clients start calling this few times a second. Search APIs can surely come later...

**Review by agorajek — APPROVED** [review]
Awesome. Just one suggestions on the pic.

File: docs/static/datajunction-illustration.png
Comment (samredai) [reply docs/static/datajunction-illustration.png:1]: Woah, love this idea. Going to update it now!

File: client/python/datajunction/client.py (hunk 1 above)
Comment (samredai) [reply client/python/datajunction/client.py:307]: From what I can see, this largely depends on the backend database you're using. That being said, I have wondered if we can add some very light in-memory caching of this to at least cache intrasecond requests, but we have to think about the read-your-write behavior there and not make it confusing. I'll run some more performance tests soon and will think about this a bit more.

**Pull request merged**